docs/ref/validators.txt

```diff
@@ -125,6 +125,15 @@ to, or in lieu of custom ``field.clean()`` methods.
     :param code: If not ``None``, overrides :attr:`code`.
     :param whitelist: If not ``None``, overrides :attr:`whitelist`.
 
+    ``EmailValidator`` doesn't aim for strict conformance to specifications.
```

Comment:
I'd omit this paragraph for now, close ticket #27029 with this commit, and leave the other one about using HTML5 validation open for further investigation.